src/analysis/functions.rs

```diff
@@ -206,14 +210,32 @@ fn analyze_function(
     configured_functions: &[&config::functions::Function],
     imports: &mut Imports,
 ) -> Info {
-    let async = func.parameters.iter().any(|parameter| parameter.scope == ParameterScope::Async);
+    let async = func.name.ends_with("_async") ||
+                func.parameters.iter()
+                               .any(|parameter| parameter.scope == ParameterScope::Async &&
+                                                parameter.c_type == "GAsyncReadyCallback");
```

Comment:
We can remove the name-ends-on-async part here now. Checking the scope and the callback type should be sufficient and more robust (-> see GDbus not using _async as a function suffix).

Also I assume this falls apart if the function is async but also has a non-gio-async callback? Does that exist?

Reply:
> Also I assume this falls apart if the function is async but also has a non-gio-async callback? Does that exist?

I don't know. This is a good question.

Reply:
You actually do know :) Elsewhere there's code that is making it an error if this case ever happens, so all fine here